File: 프로그래머스/1/12940. 최대공약수와 최소공배수/최대공약수와 최소공배수.java
class Solution {
    public int[] solution(int n, int m) {
        int[] answer = new int[2];
        int a = Math.max(n,m);
        int b = Math.min(n,m);
        
        answer[0] = gcd(a,b);
        answer[1] = lmc(a,b);
        
        return answer;
    }
    int gcd(int a, int b){
        if(b==0) return a;
        return gcd(b,a%b);
    }
    int lmc(int a, int b){
        return a*b / gcd(a,b);
    }
}

File: 프로그래머스/1/12940. 최대공약수와 최소공배수/최대공약수와 최소공배수.js
function solution(n, m) {
    return [최대공약수(n,m), 최소공배수(n,m)];
}
function 최대공약수(n,m){
    if(m == 0) return n;
    
    return 최대공약수(m,n%m);
}
function 최소공배수(n,m){
    
    return n*m/최대공약수(n,m);
}